src/pages/workspace/accounting/intacct/import/SageIntacctAddUserDimensionPage.tsx

```diff
@@ -0,0 +1,96 @@
+
+function SageIntacctAddUserDimensionPage({policy}: WithPolicyProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const policyID = policy?.id ?? '-1';
+    const userDimensions = policy?.connections?.intacct?.config?.mappings?.dimensions;
+
+    const validate = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.SAGE_INTACCT_DIMENSION_TYPE_FORM>) => {
+            const errors: FormInputErrors<typeof ONYXKEYS.FORMS.SAGE_INTACCT_DIMENSION_TYPE_FORM> = {};
+
+            if (!values[INPUT_IDS.INTEGRATION_NAME]) {
+                ErrorUtils.addErrorMessage(errors, INPUT_IDS.INTEGRATION_NAME, translate('common.error.fieldRequired'));
+            }
+
+            if (userDimensions?.some((userDimension) => userDimension.name === values[INPUT_IDS.INTEGRATION_NAME])) {
+                ErrorUtils.addErrorMessage(errors, INPUT_IDS.INTEGRATION_NAME, translate('workspace.intacct.dimensionExists'));
+            }
+
+            if (!values[INPUT_IDS.DIMENSION_TYPE]) {
+                ErrorUtils.addErrorMessage(errors, INPUT_IDS.DIMENSION_TYPE, translate('common.error.fieldRequired'));
+            }
+            return errors;
+        },
+        [translate, userDimensions],
+    );
+
+    return (
+        <ConnectionLayout
+            displayName={SageIntacctAddUserDimensionPage.displayName}
+            headerTitle="workspace.intacct.addUserDefinedDimension"
+            accessVariants={[CONST.POLICY.ACCESS_VARIANTS.ADMIN, CONST.POLICY.ACCESS_VARIANTS.PAID]}
+            policyID={policyID}
+            featureName={CONST.POLICY.MORE_FEATURES.ARE_CONNECTIONS_ENABLED}
+            contentContainerStyle={styles.flex1}
+            titleStyle={styles.ph5}
+            connectionName={CONST.POLICY.CONNECTIONS.NAME.SAGE_INTACCT}
+        >
+            <FormProvider
+                style={[styles.flexGrow1, styles.ph5]}
+                formID={ONYXKEYS.FORMS.SAGE_INTACCT_DIMENSION_TYPE_FORM}
+                validate={validate}
+                onSubmit={(value) => {
+                    addSageIntacctUserDimensions(policyID, value[INPUT_IDS.INTEGRATION_NAME], value[INPUT_IDS.DIMENSION_TYPE], userDimensions ?? []);
+                    Navigation.goBack(ROUTES.POLICY_ACCOUNTING_SAGE_INTACCT_USER_DIMENSIONS.getRoute(policyID));
+                }}
+                submitButtonText={translate('common.confirm')}
+                enabledWhenOffline
+                shouldValidateOnBlur
+                shouldValidateOnChange
+            >
+                <View style={styles.mb4}>
+                    <InputWrapper
+                        InputComponent={TextInput}
+                        inputID={INPUT_IDS.INTEGRATION_NAME}
+                        label={translate('workspace.intacct.integrationName')}
+                        aria-label={translate('workspace.intacct.integrationName')}
+                        role={CONST.ROLE.PRESENTATION}
+                        spellCheck={false}
```

Comment:
I think we should also implement auto-focus for the input here, as we did that for others page inside the App.

```javascript
const {inputCallbackRef} = useAutoFocusInput();
```

```suggestion
                        spellCheck={false}
                        ref={inputCallbackRef}
```